src/genpy/generate_initpy.py

```diff
@@ -65,19 +72,19 @@ def write_module(basedir, generated_modules):
     if not os.path.exists(basedir):
         os.makedirs(basedir)
     elif not os.path.isdir(basedir):
-        raise MsgGenerationException("file preventing the creating of module directory: %s"%dir)
+        raise MsgGenerationException('file preventing the creating of module directory: %s' % basedir)
```

Comment:
This change actually fixes a bug. The argument for the error message was supposed to be `basedir` instead of the builtin `dir()`.